### PR TITLE
Fix interval encoding to allow 0s and avoid extra spaces

### PR DESCRIPTION
**Pull request by exekias**
Fix a bugs introduced by 01d649b, also add some tests

closes https://github.com/jackc/pgx/issues/2034

### CODE REVIEW

File: pgtype/interval.go [hunk 1]
```diff
@@ -132,29 +132,25 @@ func (encodePlanIntervalCodecText) Encode(value any, buf []byte) (newBuf []byte,
 
 	if interval.Days != 0 {
 		buf = append(buf, strconv.FormatInt(int64(interval.Days), 10)...)
-		buf = append(buf, " day"...)
+		buf = append(buf, " day "...)
```
Comment (obs-gh-patrickscott) [inline pgtype/interval.go:135]: If months is 0, this will produce " day hh:mm:ss"

Comment (exekias) [reply pgtype/interval.go:135]: nope, the number goes just before. There is a test with 0 months that looks good: https://github.com/jackc/pgx/pull/2035/files#diff-deeaa7f4bfb6897c6bb770fb12921f2350ce60ca3a2a5416c84a1ec23bb2b11eR151

Comment (obs-gh-patrickscott) [reply pgtype/interval.go:135]: Ah, my mistake!

**Comment by nickbruun** [comment]
I know it's a lot to ask, but can we cut a patch 5.6.1 version with this in it?

**Comment by jackc** [comment]
@nickbruun I just released v5.7.0 that includes this.

**Comment by nickbruun** [comment]
@jackc you absolutely rock, as ever 👍

**Pull request merged**